### PR TITLE
removed RPC-Async-V1-06 and extra check which is not required as part of CreateOperationAsyncResponseValidation

### PR DESCRIPTION
**Pull request by tejaswiMinnu**
Removed - RPC-Async-V1-06 rule and its corresponding files and information.
 CreateOperationAsyncResponseValidation - has a false positive which has a check-> 'final-state-via' property set to 'azure-async-operation' 
Removed the check which is not required as the part of "CreateOperationAsyncResponseValidation" with this PR.

### CODE REVIEW

File: docs/rules.md [hunk 1]
```diff
@@ -925,6 +925,12 @@ This rule is to check if the schemas used by private endpoint conform to the com
 
 Please refer to [private-endpoint-resource-schema-validation.md](./private-endpoint-resource-schema-validation.md) for details.
 
+### PropertiesTypeObjectNoDefinition
```
Comment (visingla-ms) [inline docs/rules.md:928]: Sometimes also called "Free-form Objects", if we want to add this bit in description.

**Review by visingla-ms — APPROVED** [review]

File: packages/rulesets/src/native/legacyRules/CreateOperationAsyncResponseValidation.ts [hunk 2]
```diff
@@ -45,17 +45,6 @@ rules.push({
           location: path,
         }
```
Comment (rkmanda) [inline packages/rulesets/src/native/legacyRules/CreateOperationAsyncResponseValidation.ts:46]: remove this as well

**Review by rkmanda — CHANGES_REQUESTED** [review]
:clock1:

**Review by bdefoy — APPROVED** [review]

**Review by rkmanda — APPROVED** [review]
:shipit:

**Pull request merged**